File: engine/model/src/main/scala/tools/packet/buddy/UpdateBuddyList.scala
package tools.packet.buddy

import client.BuddyListEntry
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class UpdateBuddyList(private var _buddies: java.util.Collection[BuddyListEntry]) extends PacketInput {
  def buddies: java.util.Collection[BuddyListEntry] = _buddies

  override def opcode(): SendOpcode = SendOpcode.BUDDYLIST
}

File: engine/engine-model/src/main/java/tools/packet/buddy/UpdateBuddyList.java
package tools.packet.buddy;

import java.util.Collection;

import client.BuddyListEntry;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record UpdateBuddyList(Collection<BuddyListEntry> buddies) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.BUDDY_LIST;
   }
}
